Source/RxSwift/ObservableType+Weak.swift

```diff
@@ -47,6 +49,7 @@ extension ObservableType {
                 return Observable<O.E>.error(error)  as! O
```

Comment:
Couldn't you do 

```swift
let error: O = Observable<O.E>.error(error)
return error
```

Just to satisfy the constraint?

Reply:
No, you'd still have to cast in order to have the affectation compile.